pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -149,25 +149,31 @@ func startAutoscalersController(ctx ControllerContext, c chan error) {
 // startServicesInformer starts the service informer.
 // The synchronization of the service informer is handled in this function.
 func startServicesInformer(ctx ControllerContext, c chan error) {
+
+	informer := ctx.InformerFactory.Core().V1().Services().Informer()
+
 	// Just start the shared informer, the autodiscovery
 	// components will access it when needed.
-	go ctx.InformerFactory.Core().V1().Services().Informer().Run(ctx.StopCh)
+	ctx.InformerFactory.Start(ctx.StopCh)
 
 	// Wait for the cache to sync
 	c <- SyncInformers(map[InformerName]cache.SharedInformer{
-		servicesInformer: ctx.InformerFactory.Core().V1().Services().Informer(),
+		ServicesInformer: informer,
 	})
 }
 
 // startEndpointsInformer starts the endpoints informer.
 // The synchronization of the endpoints informer is handled in this function.
 func startEndpointsInformer(ctx ControllerContext, c chan error) {
+
+	informer := ctx.InformerFactory.Core().V1().Endpoints().Informer()
+
 	// Just start the shared informer, the autodiscovery
 	// components will access it when needed.
-	go ctx.InformerFactory.Core().V1().Endpoints().Informer().Run(ctx.StopCh)
+	ctx.InformerFactory.Start(ctx.StopCh)
```

Comment:
Is it always the same ctx.InformerFactory that we start in the different methods: `startEndpointsInformer ` `startServicesInformer`...? 
If yes, it should be done outside those functions.

Reply:
Yes it's always the same factory. However it's also started here https://github.com/DataDog/datadog-agent/blob/cf0550629056d6a9cc90249892f92f9fff2a35af/pkg/util/kubernetes/apiserver/controllers.go#L105.

Reply:
So I'm a bit unclear about the pattern here on those two handlers, should I just create a map of informers in the context, populate it from `startEndpointsInformer` & `startServicesInformer` then call `WaitForCacheSync` right after the common factory start above?

Reply:
Like this: https://github.com/DataDog/datadog-agent/commit/43de23ac73523b2f92c7e257ce8039f4522b708f

Reply:
Yes